bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java

```diff
@@ -0,0 +1,142 @@
+
+    private @NonNullByDefault({}) POP3IMAPConfig config;
+    private @Nullable ScheduledFuture<?> refreshTask;
+    private String provider = "imap";
+
+    public POP3IMAPHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(POP3IMAPConfig.class);
+
+        provider = thing.getThingTypeUID().getId(); // pop3 or imap
+
+        if (config.security == ServerSecurity.SSL) {
+            provider.concat("s");
+        }
+
+        if (config.port == 0) {
+            switch (provider) {
+                case "imap":
+                    config.port = 143;
+                    break;
+                case "imaps":
+                    config.port = 993;
+                    break;
+                case "pop3":
+                    config.port = 110;
+                    break;
+                case "pop3s":
+                    config.port = 995;
+                    break;
+                default:
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+                    return;
+            }
+        }
+
+        refreshTask = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refresh, TimeUnit.SECONDS);
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        if (refreshTask != null) {
+            if (!refreshTask.isCancelled()) {
+                refreshTask.cancel(true);
+            }
+        }
+    }
+
+    private void refresh() {
+        Session session = Session.getDefaultInstance(new Properties(), null);
+        try {
+            Store store = session.getStore(provider);
+            store.connect(config.hostname, config.port, config.username, config.password);
+
+            for (Channel channel : thing.getChannels()) {
+                if (CHANNEL_TYPE_UID_FOLDER_MAILCOUNT.equals(channel.getChannelTypeUID())) {
+                    final POP3IMAPChannelConfig channelConfig = channel.getConfiguration()
+                            .as(POP3IMAPChannelConfig.class);
+                    final String folderName = channelConfig.folder;
+                    if (folderName == null || folderName.isEmpty()) {
+                        logger.info("missing or empty folder name in channel {}", channel.getUID());
+                    } else {
+                        Folder mailbox = store.getFolder(folderName);
+                        mailbox.open(Folder.READ_ONLY);
+                        int mailNum;
+                        if (channelConfig.type == MailCountChannelType.TOTAL) {
+                            mailNum = mailbox.getMessageCount();
+                        } else {
+                            mailNum = mailbox.search(new FlagTerm(new Flags(Flags.Flag.SEEN), false)).length;
+                        }
+                        mailbox.close();
```

Comment:
Is it bad this isn't called when an exception might happen between open and this close call? otherwise put it in a finally block.